mkconcore.py

```diff
@@ -45,7 +45,13 @@
 
 prefixedgenode = ""
 sourcedir = sys.argv[2]
-outdir = sys.argv[3]
+apikeyindex = sourcedir.find('_')

```

Comment:
My suggestion is to avoid changing anything in this file.


The rationale goes like this:
The API Key implementation is futuristic (not current). It is created when invoked through the front-end. The API Keys are interpreted by Kong and passed along.

So, when a call is initiated with the correct API key, it is mapped with the correct folder. 
When the API key is wrong, Kong drops the call. When the API Key is correct (as confirmed by Kong), it accepts the call and sends the call to the FRI backend (together with the API Key), which is used to create the folder ("build"), and invoke the folder ("debug", "run", ...).

When Kong is not present, the call succeeds (with the API Key, regardless of whether the API key is valid or not). This is what happening now, when you pass the hard-coded value "xyz."

So, the proposed changes I have for this PR is:
1. Can you make a PR without changes to mkconcore.py? I believe your code should work as-is, with just the changes to main.py and test.py. API Keys are interpreted by Kong, and we shouldn't have to interpret it like this with Windows vs. Linux.
2. If someone gives an API key "", the code should still work. But without appending "_"

So an additional check to ensure the input "apikey" value is not empty before appending "_" + apikey.
If it is empty, just do not add anything. 

This will still work correctly because:
* when Kong is present and configured to expect API Keys (future status), it will drop the calls without the API Key.
* when kong is not present (current status), it will accept the calls with or without the API Key.

Reply:
> My suggestion is to avoid changing anything in this file.
> 
> The rationale goes like this: The API Key implementation is futuristic (not current). It is created when invoked through the front-end. The API Keys are interpreted by Kong and passed along.
> 
> So, when a call is initiated with the correct API key, it is mapped with the correct folder. When the API key is wrong, Kong drops the call. When the API Key is correct (as confirmed by Kong), it accepts the call and sends the call to the FRI backend (together with the API Key), which is used to create the folder ("build"), and invoke the folder ("debug", "run", ...).
> 
> When Kong is not present, the call succeeds (with the API Key, regardless of whether the API key is valid or not). This is what happening now, when you pass the hard-coded value "xyz."
> 
> So, the proposed changes I have for this PR is:
> 
> 1. Can you make a PR without changes to mkconcore.py? I believe your code should work as-is, with just the changes to main.py and test.py. API Keys are interpreted by Kong, and we shouldn't have to interpret it like this with Windows vs. Linux.
> 2. If someone gives an API key "", the code should still work. But without appending "_"
> 
> So an additional check to ensure the input "apikey" value is not empty before appending "_" + apikey. If it is empty, just do not add anything.
> 
> This will still work correctly because:
> 
> * when Kong is present and configured to expect API Keys (future status), it will drop the calls without the API Key.
> * when kong is not present (current status), it will accept the calls with or without the API Key.

**Not For Kong**
I understand what is being conveyed, but if no changes in mkconcore.py are entertained then it will not be possible to create out_directory of name 'graphmlfilename_apikey' in case user provides api key. If this idea of making folder by the name format  'graphmlfilename_apikey'  has to be dropped, then there is no use of passing apikeys to functions (debug, run, stop, clear etc.). Build certainly needs api key if passed which is the neutral case.

So the main concern is whether or not api keys are passed in debug, run, clear, stop, download etc., anyway they will perform their respective functions. The only use I find is that if the out_dir with name of graphml file is attached with apikey (eg. sample_xyz which can only be created by making changes in mkconcore.py) and we need to access it, then passing apikey is beneficial.  (For Linux, Windows etc.)

**For Kong**
When it comes to Kong, apikeys must be passed. Therefore all changes conveyed by you are perfect for now, and they will become more clear after Kong configuration will be completed. Therefore, I will be making changes as proposed above.